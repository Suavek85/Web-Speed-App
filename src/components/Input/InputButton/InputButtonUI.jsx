import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

//INTERNAL
import { appStates } from '../../../constants/states'
import { btnsContent } from '../../../constants/content'
import './InputButtonUI.scss'  

function InputButtonUI(props) {
 
  const { getAppState, handleGetReportClick } = props

  //BUTTON FORMATTING - CLASSES AND CONTENT
  const isStatus = state => getAppState === state
  
  const getContentFromStatus = state => { 
    const contentIndex = Object.keys(btnsContent).findIndex(el => el === state)
    return Object.values(btnsContent)[contentIndex]
  }

  const getButtonClasses = cls({
    'inputbutton__wrapper': true,
    'inputbutton__wrapper--inactive': isStatus(appStates.INACTIVE),
    'inputbutton__wrapper--loading': isStatus(appStates.LOADING),
    'inputbutton__wrapper--error': isStatus(appStates.ERROR),
    'inputbutton__wrapper--success': isStatus(appStates.SUCCESS),
  })

  return (
    <>
      <button 
        onClick={handleGetReportClick} 
        disabled={isStatus(appStates.LOADING)}
        className={getButtonClasses}
      >   
        {getContentFromStatus(getAppState)}
      </button>
    </>
  )
}

InputButtonUI.propTypes = {
  data: PropTypes.string,
  getAppState: PropTypes.string,
  handleGetReportClick: PropTypes.func
}

export default InputButtonUI