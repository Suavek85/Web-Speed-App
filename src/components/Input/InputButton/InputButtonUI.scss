@import "../../../scss/styles.scss";

.inputbutton__wrapper {
    display: block;
    padding: 16px;
    margin-bottom: 20px;
    color: white;
    outline: none;
    border: none;
    cursor: pointer;
    &--inactive {
      background-color: rgba(55,71,133,.8);
      &:active {
        transform: translateY(2px);
        background-color:rgba(55,71,133,1)
      }
    }
    &--loading {
        background-color: $color-warning-loading;
    }
    &--error {
      background-color: $color-bottom-background;
    }
    &--success {
      background-color: $color-green-success;
    }
}

:export {
  block: inputbutton__wrapper;
  blockInactive: inputbutton__wrapper--inactive;
  blockLoading: inputbutton__wrapper--loading;
  blockError: inputbutton__wrapper--error;
  blockSuccess: inputbutton__wrapper--success;
}