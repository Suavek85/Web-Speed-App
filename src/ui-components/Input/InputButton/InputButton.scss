.inputbutton__wrapper {
    display: block;
    padding: 12px;
    margin-bottom: 20px;
    background-color:rgba(55,71,133,.8);
    color: #A8D0E6;
    outline: none;
    border: none;
    &:active {
        transform: translateY(2px);
        background-color:rgba(55,71,133,1)
    }
 
}