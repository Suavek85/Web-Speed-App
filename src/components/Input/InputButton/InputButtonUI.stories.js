import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, select } from "@storybook/addon-knobs"
import InputButtonUI from './InputButtonUI'
import { appStates } from '../../../constants/states'

const stories = storiesOf('UI Components', module)
stories.addDecorator(withKnobs)

stories.add('Input Button', () => {
  const groupId = 'Props'
  const statusesKnob = select('Button states', appStates, Object.values(appStates)[0], groupId)
  return <InputButtonUI getAppState={statusesKnob} />
})