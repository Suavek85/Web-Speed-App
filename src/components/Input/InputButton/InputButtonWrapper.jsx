import React from 'react'
import { useSelector, useDispatch } from "react-redux"
import axios from 'axios'

//INTERNAL
import InputButtonUI from './InputButtonUI'
import { appStates } from '../../../constants/states'
import firebase from './../../../firestore'
import getState from '../../../actions/stateActions.js'

function InputButtonWrapper() {

  //FETCHING API KEY FROM FIRESTORE DB
  let key = ''
  const db = firebase.firestore()
  const docRef = db.collection("key").doc("maAkzowoP49Eu5OKNQV7")
  docRef.get().then(doc => { if(doc.exists) key = doc.data().keyid })
 
  //FETCHING API

  //useSelector to take url from store

  const getUrlState = useSelector(state => state.urlReducer.getUrl)
  const inputUrl = getUrlState

  const url = `https://www.googleapis.com/pagespeedonline/v5/runPagespeed?url=${inputUrl}&key=${key}`

  const dispatch = useDispatch()
 
  async function makeGetRequest() {
    try {
      let res = await axios.get(url)
      let allData = res.data      
      dispatch(getState(allData.loadingExperience.metrics.FIRST_CONTENTFUL_PAINT_MS.category))
    } catch(err) {
      dispatch({ type: appStates.ERROR })
      setTimeout(function () {
        dispatch({ type: appStates.INACTIVE })
      }, 4000)
    }
  }

  const handleGetReportClick = () => { 
    dispatch({ type: appStates.LOADING })
    makeGetRequest() 
  }
  const getAppState = useSelector(state => state.stateReducer.getAppState)

  return (
    <InputButtonUI 
      handleGetReportClick={handleGetReportClick} 
      getAppState={getAppState} 
    />
  )
}

export default InputButtonWrapper