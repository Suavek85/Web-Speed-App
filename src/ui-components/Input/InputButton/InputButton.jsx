import React from 'react'
import PropTypes from 'prop-types'
import { appStates } from '../../../constants/states'
import { GETREPORT, LOADING } from '../../../constants/content'
import './InputButton.scss'
import cls from 'classnames'

function InputButton(props) {
  const {getReportStatus, handleGetReportClick } = props
  const loadingStatus = getReportStatus === appStates.LOADING
  const inactiveStatus = getReportStatus === appStates.INACTIVE
  const getButtonClasses = cls({
    'inputbutton__wrapper': true,
    'inputbutton__wrapper--inactive': inactiveStatus,
    'inputbutton__wrapper--loading': loadingStatus,
  })
  return (
    <button 
      onClick={handleGetReportClick} 
      disabled={loadingStatus}
      className={getButtonClasses}
    >   
      {inactiveStatus && GETREPORT}
      {loadingStatus && LOADING}
    </button>
  )
}

InputButton.propTypes = {
  getReportStatus: PropTypes.string.isRequired,
  handleGetReportClick: PropTypes.func.isRequired
}

InputButton.defaultProps = {
  getReportStatus: appStates.INACTIVE,
  handleGetReportClick: () => {}
}

export default InputButton