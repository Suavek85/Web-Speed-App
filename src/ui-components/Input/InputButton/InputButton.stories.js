import React from 'react';
import InputButton from './InputButton';

export default {
    title: 'Input Button',
  };

export const InputBtn = () => <InputButton />;