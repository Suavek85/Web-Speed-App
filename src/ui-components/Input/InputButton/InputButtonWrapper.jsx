import React from 'react'
import InputButton from './InputButton'

function InputButtonWrapper() {
  return (
    <InputButton />
  )
}

export default InputButtonWrapper