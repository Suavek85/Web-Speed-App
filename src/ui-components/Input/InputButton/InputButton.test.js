import React from 'react'
import { shallow } from 'enzyme'
import InputButton from './InputButton'

it('Should match the snapshot', () => {
  const wrapper = shallow(<InputButton />).debug()
  expect(wrapper).toMatchSnapshot()
})
