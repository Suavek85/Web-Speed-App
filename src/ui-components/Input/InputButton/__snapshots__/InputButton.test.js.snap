// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
"<button onClick={[Function: handleGetReportClick]} disabled={false} className=\\"inputbutton__wrapper inputbutton__wrapper--inactive\\">
  Get report
</button>"
`;
